[cinder] Migrate policy.json to policy.yaml

hemna opened the pull request:
For cinder Queens and on, Cinder can use a policy.yaml file
instead of a policy.json file.  The policy.yaml is the default
for Rocky release and forward.

The rocky release notes describes the upgrade.
https://docs.openstack.org/releasenotes/cinder/rocky.html

hemna commented:
This patch is a pre-emptive migration for Train upgrade.

joker-official reviewed (commented):
Did you manually compare the rules and copy over our stuff to yaml or how did you convert the file?

hemna commented:
As described in the release notes, the yaml file is basically. identical minus the wrapping {} around the json, and removing
of the , at the end of each line.   

So I basically did this:
cp policy.json policy.yaml
edit the file, removed the {} and , 

Then I generated the default cinder-policy.yaml.sample file from tox -egenpolicy
and compared the policy entries.  I changed some of the invalid policy entries that we had for
managing volume attachments, such as 
"volume:attach" : "...."  to the correct entry using the volume_extension actions.
I also added the group actions which are replacements for the old consistency group entries as CG's as an API don't exist anymore.

joker-official commented on openstack/cinder/templates/etc/_cinder-policy.yaml.tpl:
```diff
@@ -0,0 +1,143 @@
+"context_is_viewer":  "rule:context_is_editor or rule:viewer"
+"view_all": "role:member or role:volume_viewer or role:volume_admin or role:cloud_volume_admin"
+"edit_all": "role:member or role:volume_admin or role:cloud_volume_admin"
+"default": "rule:context_is_admin"
```
Release-Notes state, that `default` is not used inside the code anymore, if I read this correctly.

joker-official commented on openstack/cinder/templates/etc/_cinder-policy.yaml.tpl:
```diff
@@ -0,0 +1,143 @@
+"group:list_replication_targets" :   "rule:context_is_admin"
+
+# List all backend pools
+"scheduler_extension:scheduler_stats:get_pools" : "rule:context_is_admin
```
This line is missing a closing `"`.

joker-official commented:
Nice explanation, thanks.

joker-official commented on openstack/cinder/templates/etc/_cinder-policy.yaml.tpl:
```diff
@@ -0,0 +1,142 @@
+"group:reset_group_snapshot_status" :   "rule:context_is_admin"
+"group:enable_replication" :   "rule:context_is_admin"
+"group:disable_replication" :   "rule:context_is_admin"
+"group:failover_replication" :   "rule:context_is_admin"
+"group:list_replication_targets" :   "rule:context_is_admin"
```
hm ... I'm annoyed by those lines having a space before the `:`. We don't do it everywhere in the file. Can we straighten this out with a small run of `sed -r 's/" +:/":/' _cinder-policy.yaml.tpl`, please?
